include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -1070,12 +1070,46 @@ template <> class ContiguousInternalMemoryAlgorithmDataFacade<MLD> : public Algo
         InitializeInternalPointers(allocator->GetLayout(), allocator->GetMemory());
     }
 
-    const partition::MultiLevelPartitionView &GetMultiLevelPartition() const override
+    LevelID GetQueryLevel(NodeID source, NodeID target, NodeID node) const override
     {
-        return mld_partition;
+        return mld_partition.GetQueryLevel(source, target, node);
     }
 
-    const partition::CellStorageView &GetCellStorage() const override { return mld_cell_storage; }
+    LevelID GetHighestDifferentLevel(NodeID first, NodeID second) const override
+    {
+        return mld_partition.GetHighestDifferentLevel(first, second);
+    }
+
+    void ForEachSourceNodes(LevelID level, NodeID node, NodeFnT const &fn) const override
```

Comment:
Functions like these are great for the convenience but I'm a little bit worried about introducing interfaces that use `std::function` objects.

I would be great if we could fine an interface that doesn't need this and still makes it possible to swap out enough so that you can implement your external algorithms.

/cc @oxidase might have some ideas as well.